docs/lang/articles/basic/field.md

```diff
@@ -14,42 +14,170 @@ A 0D (zero-dimensional) field contains *only* one element.
 
 ## Scalar fields
 
-Scalar fields refer to the fields that store scalars and are the most basic fields. A 0D scalar field is a single scalar.
+Scalar fields refer to the fields that store scalars and are the most basic fields. A 0D scalar field is a single scalar; a 1D scalar field is a 1D array of scalars; a 2D scalar field is a 2D array of scalars, and so on and so forth.
 
 
 ### Declaration
 
-```python
-import taichi as ti
-ti.init(arch=ti.cpu)
+The simplest way to declare a scalar field is to call `ti.field(dtype, shape)`, where `dtype` is a primitive data type as explained in [type system](../type_system/type.md) and `shape` is a tuple of integers.
+
++ To declare a 0D scalar field, set its shape to the empty tuple `()`.
+
+    ```python
+    # Declare a 0D scalar field whose data type is f32
+    f_0d = ti.field(ti.f32, shape=())  # 0D field
+    ```
+
+    The memory layout of the field `f_0d` will be like:
+
+    ```
+        ┌─────┐
+        │     │
+        └─────┴
+        └─────┘
+     f_0d.shape=()
+    ```
+
++ To declare a 1D scalar field of length `n`, set its shape to `n` or `(n,)`:
+
+    ```python
+    f_1d = ti.field(ti.i32, shape=9)  # 1D field
+    ```
+
+    The memory layout of the field `f_1d` will be like:
+
+    ```
+    ┌───┬───┬───┬───┬───┬───┬───┬───┬───┐
+    │   │   │   │   │   │   │   │   │   │
+    └───┴───┴───┴───┴───┴───┴───┴───┴───┘
+    └───────────────────────────────────┘
+            f_1d.shape = (9,)
+    ```
+
++ To declare a 2D scalar field, set the size of its first two dimensions, i.e. the number of rows and columns, respectively. For example, the following code defines a 2D scalar field of shape (3, 6), which has 3 rows and 6 columns:
+
+    ```python
+    f_2d = ti.field(int, shape=(3, 6))  # 2D field
+    ```
+
+    The memory layout of the field `f_2d` will be like:
```

Comment:
```suggestion
    An illustration of f_2d is below:
```
Let's avoid using `memory layout` here, which has a different meaning. Just use `An illustration of f_2d is below` is good enough.